File: fig4.py
import numpy as np
import matplotlib.pyplot as plt

fname = ['fig4_K3.txt', 'fig4_K3z.txt',
         'fig4_Kc.txt', 'fig4_Kcz.txt',
         'fig4_K5.txt', 'fig4_K5z.txt']
title = [r'$K = 0.3$', 'zoomed out',
         r'$K = 0.4407$', 'zoomed out',
         r'$K = 0.5$', 'zoomed out']

plt.figure(figsize=(4,6))

for i,fname in enumerate(fname):
    s = np.loadtxt(fname)
    plt.subplot(3,2,i+1)
    plt.title(title[i])
    plt.imshow(s, cmap='gray')
    plt.xticks([])
    plt.yticks([])

plt.tight_layout()
plt.savefig('fig4.eps')
plt.show()

File: fig4.cpp
#include "IsingMC.h"
#include<cmath>

main() {
    long L(128);
    double Kc(log(1+sqrt(2))/2);
    IsingMC m(L);

    m.set_K(0.3, 1<<14);
    m.write("fig4_K3.txt");
    m.zoom_out();
    m.write("fig4_K3z.txt");

    m.set_K(Kc, 1<<14);
    m.write("fig4_Kc.txt");
    m.zoom_out();
    m.write("fig4_Kcz.txt");
    
    m.set_K(0.5, 1<<14);
    m.write("fig4_K5.txt");
    m.zoom_out();
    m.write("fig4_K5z.txt");
}